addon/services/user-agent.js

```diff
@@ -1,106 +1,135 @@
-import { assign } from '@ember/polyfills';
-import { isEqual } from '@ember/utils';
-import { assert } from '@ember/debug';
+import { computed, get, set } from '@ember/object';
 import { getOwner } from '@ember/application';
+import { deprecate } from '@ember/application/deprecations';
 import UAParser from 'ua-parser-js';
 
 export default Service.extend({
+  _parser: computed(() => new UAParser),
 
-  fastboot: computed(function() {
-    return getOwner(this).lookup('service:fastboot');
-  }),
+  parser: computed('extensions', {
+    get() {
+      deprecate(
+        'Usage of the private property `parser` is deprecated. If you need to fiddle around with this private property, please open an issue and we can discuss.',
+        false,
+        { id: 'ember-useragent.service.parser', until: '1.0.0' }
+      );
 
-  isFastBoot: readOnly('fastboot.isFastBoot'),
+      return get(this, '_parser');
+    }
+  }).readOnly(),
 
-  userAgent: computed(function() {
-    if (get(this, 'isFastBoot')) {
-      let headers = get(this, 'fastboot.request.headers');
-      let userAgent = headers.get('user-agent');
+  UAParser: computed(function () {
+    deprecate(
+      'Usage of the property `UAParser` is deprecated. To get the UAParser class, import it as `import UAParser from \'ua-parser-js\';`. To get the instance of that class used by this service, as this property previously incorrectly returned, get the `parser` property.',
+      false,
+      { id: 'ember-useragent.service.UAParser', until: '1.0.0' }
+    );
 
-      assert('No userAgent present in ember-useragent/services/user-agent (FastBoot)', userAgent);
-      return userAgent;
-    } else {
-      if (window && window.navigator) {
-        let userAgent = window.navigator.userAgent;
+    return get(this, '_parser');
+  }),
 
-        assert('No userAgent present in ember-useragent/services/user-agent (Browser)', userAgent);
-        return userAgent;
-      }
+  userAgent: computed({
+    get() {
+      const parser = get(this, '_parser');
+      return parser.getUA();
+    },
+    set(key, value) {
+      const parser = get(this, '_parser');
+      parser.setUA(value);
+      set(this, '_parser', parser);
+
+      return value;
     }
   }),
 
-  UAParser: computed('userAgent', function() {
-    let userAgent = get(this, 'userAgent');
+  fastboot: computed(function () {
+    deprecate(
+      'Usage of the private property `fastboot` is deprecated. Inject the `fastboot` service yourself instead.',
+      false,
+      { id: 'ember-useragent.service.fastboot', until: '1.0.0' }
+    );
 
-    if (get(this, 'isFastBoot')) {
-      let UAParser = FastBoot.require('ua-parser-js');
-      return new UAParser(userAgent);
-    }
+    return getOwner(this).lookup('service:fastboot');
+  }),
+
+  isFastBoot: computed(function () {
+    deprecate(
+      'Usage of the private property `isFastBoot` is deprecated. Inject the `fastboot` service yourself and get `fastboot.isFastBoot` instead.',
```

Comment:
All the Fastboot-specific logic is now in an instance-initializer. So these two properties aren't needed any more. Since some folks might depend on it, I deprecated them instead of removing directly.